tooling/reproducible/macos_repro_build_compare.sh

```diff
@@ -0,0 +1,673 @@
+      fixed_param="$split_param"
+      fixed_value="=$MAC_SDK_LOCATION " ;
+    fi
+    if [[ "$fixed_param" == "--openjdk-target="* ]]; then
+      IFS='=' read -r split_param split_value <<< "$fixed_param"
+      fixed_param="$split_param"
+      fixed_value="=$split_value "
+    fi
+    if [[ "$fixed_param" == "--with-version-opt="* ]]; then
+      IFS='=' read -r split_param split_value <<< "$fixed_param"
+      fixed_param="$split_param"
+      fixed_value="=$split_value"
+    fi
+    if [ "$fixed_param" == "--tag" ]; then fixed_param="-b" fixed_value=" $fixed_value" ; fi
+    if [ "$fixed_param" == "--target-file-name" ]; then target_file="$fixed_value" ; fixed_value=" $fixed_value" ; fi
+    if [ "$fixed_param" == "--freetype-dir" ]; then fixed_value=" $fixed_value" ; fi
+    if [ "$fixed_param" == "--user-openjdk-build-root-directory" ]; then fixed_value=" $WORK_DIR/temurin-build/workspace/build/openjdkbuild/ " ; fi
+    if [ "$fixed_param" == "--build-variant" ]; then fixed_value=" $fixed_value " ; fi
+    if [ "$fixed_param" == "--build-reproducible-date" ]; then fixed_value=" $fixed_value" ; fi
+
+    if containsElement "$fixed_param" "${CONFIG_ARGS[@]}"; then
+      # Check if fixed_param is in CONFIG_ARGS
+      STRINGTOADD="$fixed_param$fixed_value"
+      CONFIG_ARRAY+=("$STRINGTOADD")
+    elif containsElement "$fixed_param" "${NOTUSE_ARGS[@]}"; then
+      # Check if fixed_param is in NOTUSE_ARGS
+      STRINGTOADD="$fixed_param$fixed_value"
+      IGNORED_ARRAY+=("$STRINGTOADD")
+    elif containsElement "$fixed_param" "${FINAL_ARG[@]}"; then
+      # Check if parameter should be at the end, usually just version & variant
+      STRINGTOADD="$fixed_param$fixed_value"
+      FINAL_ARRAY+=("$STRINGTOADD")
+    else
+      # Neither Config Or Ignore, so Build
+      STRINGTOADD="$fixed_param$fixed_value"
+      BUILD_ARRAY+=("$STRINGTOADD")
+    fi
+  done
+
+  # Construct Final Parameter String
+  for element in "${BUILD_ARRAY[@]}"; do
+    build_string+="$element"
+  done
+
+  for element in "${CONFIG_ARRAY[@]}"; do
+    config_string+="$element"
+  done
+
+  for element in "${FINAL_ARRAY[@]}"; do
+    final_string+="$element"
+  done
+
+  final_params="$build_string --configure-args \"$config_string\" $final_string"
+
+echo ""
+  echo "Make JDK Any Platform Argument List = "
+  echo "$final_params"
+  echo ""
+  echo "Parameters Parsed Successfully"
+}
+
+Build_JDK() {
+  echo "Building JDK..."
+
+  # Trigger Build
+  cd "$WORK_DIR"
+  echo "cd temurin-build && ./makejdk-any-platform.sh $final_params 2>&1 | tee build.$$.log" | sh
+  # Copy The Built JDK To The Working Directory
+  cp $WORK_DIR/temurin-build/workspace/target/"$target_file" $WORK_DIR/built_jdk.tar.gz
+}
+
+Compare_JDK() {
+  echo "Comparing JDKs"
+  echo ""
+  mkdir "$WORK_DIR/compare"
+  cp "$WORK_DIR/src_jdk_dist.tar.gz" "$WORK_DIR/compare"
+  cp "$WORK_DIR/built_jdk.tar.gz" "$WORK_DIR/compare"
+
+  # Get The Current Versions Of The Reproducible Build Scripts
```

Comment:
Possibly ought to integrate some of Sophia's latest updates to the Compare_JDK() function - for Windows : https://github.com/adoptium/temurin-build/commit/ad4a1c3ae345a6f9b721b5fbea9ebfc0f1005a83 ?

Reply:
Done, aligned those elements I think required it.